tests/api/test_core_coord_translation_wh.cpp

```diff
@@ -0,0 +1,279 @@
+// so logical coordinates should cover all physical coordinates.
+TEST(CoordinateManager, CoordinateManagerWormholeDRAMNoHarvesting) {
+    std::shared_ptr<CoordinateManager> coordinate_manager =
+        CoordinateManager::get_coordinate_manager(tt::ARCH::WORMHOLE_B0);
+
+    const size_t num_dram_banks = tt::umd::wormhole::NUM_DRAM_BANKS;
+    const size_t num_noc_ports_per_bank = tt::umd::wormhole::NUM_NOC_PORTS_PER_DRAM_BANK;
+    const std::vector<tt_xy_pair>& dram_cores = tt::umd::wormhole::DRAM_CORES;
+
+    for (size_t dram_bank = 0; dram_bank < num_dram_banks; dram_bank++) {
+        for (size_t noc_port = 0; noc_port < num_noc_ports_per_bank; noc_port++) {
+            const CoreCoord dram_logical(dram_bank, noc_port, CoreType::DRAM, CoordSystem::LOGICAL);
+            const size_t physical_core_index = dram_bank * num_noc_ports_per_bank + noc_port;
+            const CoreCoord expected_physical = CoreCoord(
+                dram_cores[physical_core_index].x,
+                dram_cores[physical_core_index].y,
+                CoreType::DRAM,
+                CoordSystem::PHYSICAL);
+
+            const CoreCoord dram_physical = coordinate_manager->to(dram_logical, CoordSystem::PHYSICAL);
+
+            EXPECT_EQ(dram_physical, expected_physical);
+        }
+    }
+}
+
+// Test that physical and virtual coordinates are the same for all logical coordinates, since there is no DRAM
+// harvesting.
+TEST(CoordinateManager, CoordinateManagerWormholeETHPhysicalEqualVirtual) {
+    std::shared_ptr<CoordinateManager> coordinate_manager =
+        CoordinateManager::get_coordinate_manager(tt::ARCH::WORMHOLE_B0);
+    const tt_xy_pair eth_grid_size = tt::umd::wormhole::ETH_GRID_SIZE;
+
+    for (size_t x = 0; x < eth_grid_size.x; x++) {
+        for (size_t y = 0; y < eth_grid_size.y; y++) {
+            const CoreCoord eth_logical = CoreCoord(x, y, CoreType::ETH, CoordSystem::LOGICAL);
+            const CoreCoord eth_virtual = coordinate_manager->to(eth_logical, CoordSystem::VIRTUAL);
+            const CoreCoord eth_physical = coordinate_manager->to(eth_logical, CoordSystem::PHYSICAL);
+
+            EXPECT_EQ(eth_virtual.x, eth_physical.x);
+            EXPECT_EQ(eth_virtual.y, eth_physical.y);
+        }
+    }
+}
+
+// Test translation of logical to translated ethernet coordinates.
+TEST(CoordinateManager, CoordinateManagerWormholeETHLogicalToTranslated) {
+    std::shared_ptr<CoordinateManager> coordinate_manager =
+        CoordinateManager::get_coordinate_manager(tt::ARCH::WORMHOLE_B0);
+    const tt_xy_pair eth_grid_size = tt::umd::wormhole::ETH_GRID_SIZE;
+
+    for (size_t x = 0; x < eth_grid_size.x; x++) {
+        for (size_t y = 0; y < eth_grid_size.y; y++) {
+            const CoreCoord eth_logical = CoreCoord(x, y, CoreType::ETH, CoordSystem::LOGICAL);
+            const CoreCoord eth_translated = coordinate_manager->to(eth_logical, CoordSystem::TRANSLATED);
+
+            EXPECT_EQ(eth_translated.x, x + 18);
```

Comment:
Can we defined these constants 18 and 16 somewhere in coord_manager/soc descriptor?

Reply:
Moved it to implementation files